suffix_update.py

```diff
@@ -0,0 +1,132 @@
+import sys
+is_py_2 = sys.version_info[0] == 2
+
+
+def read_suffix_list(content):
+    '''
+    Read and parse suffix list.
+    '''
+    # remove comment & blank line
+    suffix_list = [line.lower().strip()
+                   for line in contents
+                   if not line.startswith('//') and line.strip()]
+
+    # find idna decodable suffixs
+    idna_encoded_suffix_list = []
+    for suffix in suffix_list:
+        try:
+            encoded = suffix.encode('idna').decode('ascii')
+            if suffix != encoded:
+                idna_encoded_suffix_list.append(encoded)
+        except Exception:
+            pass
```

Comment:
Seems like you don't want to catch all the thing, but rather want to catch `UnicodeEncodeError`. Does that sound right?